src/dotnet/commands/dotnet-tool/install/ToolInstallCommand.cs

```diff
@@ -17,161 +17,97 @@
-            IReporter reporter = null)
+            ToolInstallGlobalOrToolPathCommand toolInstallGlobalOrToolPathCommand = null,
+            ToolInstallLocalCommand toolInstallLocalCommand = null)
             : base(parseResult)
         {
-            if (appliedCommand == null)
-            {
-                throw new ArgumentNullException(nameof(appliedCommand));
-            }
-
-            _packageId = new PackageId(appliedCommand.Arguments.Single());
-            _packageVersion = appliedCommand.ValueOrDefault<string>("version");
-            _configFilePath = appliedCommand.ValueOrDefault<string>("configfile");
+            _appliedCommand = appliedCommand ?? throw new ArgumentNullException(nameof(appliedCommand));
+            _parseResult = parseResult ?? throw new ArgumentNullException(nameof(parseResult));
+            _toolInstallLocalCommand =
+                toolInstallLocalCommand
+                ?? new ToolInstallLocalCommand(_appliedCommand, _parseResult);
+
+            _toolInstallGlobalOrToolPathCommand =
+                toolInstallGlobalOrToolPathCommand
+                ?? new ToolInstallGlobalOrToolPathCommand(_appliedCommand, _parseResult);
+
+            _global = appliedCommand.ValueOrDefault<bool>(GlobalOption);
+            _local = appliedCommand.ValueOrDefault<bool>(LocalOption);
+            _toolPath = appliedCommand.SingleArgumentOrDefault(ToolPathOption);
+            _toolManifestOption = appliedCommand.ValueOrDefault<string>("tool-manifest");
             _framework = appliedCommand.ValueOrDefault<string>("framework");
-            _source = appliedCommand.ValueOrDefault<string[]>("add-source");
-            _global = appliedCommand.ValueOrDefault<bool>("global");
-            _verbosity = appliedCommand.SingleArgumentOrDefault("verbosity");
-            _toolPath = appliedCommand.SingleArgumentOrDefault("tool-path");
-
-            _createToolPackageStoresAndInstaller = createToolPackageStoreAndInstaller ?? ToolPackageFactory.CreateToolPackageStoresAndInstaller;
-			_forwardRestoreArguments = appliedCommand.OptionValuesToBeForwarded();
-
-            _environmentPathInstruction = environmentPathInstruction
-                ?? EnvironmentPathFactory.CreateEnvironmentPathInstruction();
-            _createShellShimRepository = createShellShimRepository ?? ShellShimRepositoryFactory.CreateShellShimRepository;
-
-            _reporter = (reporter ?? Reporter.Output);
-            _errorReporter = (reporter ?? Reporter.Error);
         }
 
         public override int Execute()
         {
-            if (string.IsNullOrWhiteSpace(_toolPath) && !_global)
-            {
-                throw new GracefulException(LocalizableStrings.InstallToolCommandNeedGlobalOrToolPath);
-            }
+            EnsureNoConflictGlobalLocalToolPathOption();
 
-            if (!string.IsNullOrWhiteSpace(_toolPath) && _global)
+            if (_global || !string.IsNullOrWhiteSpace(_toolPath))
```

Comment:
So, just to confirm, we want the default to be global tools?

Reply:
No, it is the opposite. The intent is default to local tools and the code does that way.

This test https://github.com/dotnet/cli/pull/10312/files/873dc6820e7c2a5a4db0312940ff81442e05e402#diff-82a9c560c0912ca31d5e1f315fb312d9R162 is testing `dotnet tool install {_packageIdA.ToString()}` will run local tools.